include/tvm/runtime/device_api.h

```diff
@@ -181,7 +181,38 @@ class TVM_DLL DeviceAPI {
    * \param ptr The pointer to be freed.
    */
   virtual void FreeWorkspace(TVMContext ctx, void* ptr);
-
+  /*!
+   * \brief Allocate a two dimensional texture data space on device
+   * \param ctx The device context to perform operation.
+   * \param width The width of the 2d texture in elements
+   * \param height The height of the 2d texture in elements
+   * \param type_hint The type of elements.
+   */
```

Comment:
It would be great to figure out a way to remove the Texture related functions from the Device API interface, so we can focus on the NDArray related parts.